### PR TITLE
WT-9947 Tidy up cursor->bound API error messages

### PR DESCRIPTION
**Pull request by clarissecheah**
- Make bounded API error messages consistent.
- Update bounded testing to account for new configuration changes.
- Expand cursor_bound01 to include edge cases.

### CODE REVIEW

**Comment by luke-pearson** [comment]
I know this isn't exactly work for this PR but can we expand `test_cursor_bound01` with some edge cases:
```
// This is just testing the case where we don't provide and action and the bound is still set.
cursor->bound("bound=lower")
// This is showing the WT_PREFIX_MATCH will accept longer config strings even though they aren't correct.
cursor->bound("action=setting,bound=lower")
// This one should throw an error as "dump" isn't a valid action.
cursor->bound("action=dump")
// This is again testing WT_PREFIX_MATCH
cursor->bound("action=clearing")
// This is testing that a substring of the prefix won't match.
cursor->bound("action=cl")
```

**Comment by clarissecheah** [comment]
> I know this isn't exactly work for this PR but can we expand `test_cursor_bound01` with some edge cases:
> 
> ```
> // This is just testing the case where we don't provide and action and the bound is still set.
> cursor->bound("bound=lower")
> // This is showing the WT_PREFIX_MATCH will accept longer config strings even though they aren't correct.
> cursor->bound("action=setting,bound=lower")
> // This one should throw an error as "dump" isn't a valid action.
> cursor->bound("action=dump")
> // This is again testing WT_PREFIX_MATCH
> cursor->bound("action=clearing")
> // This is testing that a substring of the prefix won't match.
> cursor->bound("action=cl")
> ```

Added those edge cases, and one extra.

**Review by luke-pearson — APPROVED** [review]
LGTM.

File: test/suite/test_cursor_bound01.py [hunk 1]
```diff
@@ -108,7 +108,42 @@ def test_bound_api(self):
...
+        cursor.reset()
+
+        cursor.set_key(self.gen_key(1))
+        cursor.bound("action=set,bound=lower")
+        # Giving a longer config string works, WT_PREFIX_MATCH will accept it.
+        self.assertEqual(cursor.bound("action=clearing"), 0)
+        cursor.reset()
+
+        cursor.set_key(self.gen_key(1))
+        cursor.bound("action=set,bound=lower")
```
Comment (jiechenbo) [inline test/suite/test_cursor_bound01.py:137]: I think this line got duped and can be removed let me know if I'm misunderstanding otherwise

**Review by jiechenbo — APPROVED** [review]
LGTM! just one comment

File: test/suite/test_cursor_bound01.py (hunk 1 above)
Comment (clarissecheah) [reply test/suite/test_cursor_bound01.py:137]: The idea was to have a lower bound set for an "action" thats not setting the bound to then be called, but that's not been made a hard rule throughout this test, and the line can be removed.
